tests/system/model_monitoring/test_app.py

```diff
@@ -1608,3 +1608,163 @@ def test_histogram_app(self) -> None:
+            )
+            serving_fn_v2.invoke(
+                f"v2/models/{self._model_name}/infer",
+                body=json.dumps({"inputs": [[0, 0, 0, 0]] * 100}),
+            )
+
+        time.sleep(30)
+
+        # Get the model endpoints
+        model_endpoint_with_sample = mlrun.get_run_db().get_model_endpoint(
+            name=self._model_name,
+            project=self.project_name,
+            function_name=self._serving_function_name_with_sample,
+            function_tag="latest",
+        )
+
+        model_endpoint_without_sample = mlrun.get_run_db().get_model_endpoint(
+            name=self._model_name,
+            project=self.project_name,
+            function_name=self._serving_function_name_without_sample,
+            function_tag="latest",
+        )
+
+        # Validate the sampling percentage
+        assert model_endpoint_with_sample.status.sampling_percentage == 15.5
+        assert model_endpoint_without_sample.status.sampling_percentage == 100
+
+        self._test_predictions_table(
+            ep_id_with_sample=model_endpoint_with_sample.metadata.uid,
+            ep_id_without_sample=model_endpoint_without_sample.metadata.uid,
+        )
+
+    def _test_predictions_table(
+        self, ep_id_with_sample: str, ep_id_without_sample: str
+    ) -> None:
+        if self._tsdb_storage.type == mm_constants.TSDBTarget.V3IO_TSDB:
+            predictions_df: pd.DataFrame = self._tsdb_storage._get_records(
+                table=mm_constants.FileTargetKind.PREDICTIONS, start="0", end="now"
+            )
+
+        else:
+            # TDEngine
+            predictions_df: pd.DataFrame = self._tsdb_storage._get_records(
+                table=self._tsdb_storage.tables[
+                    mm_constants.TDEngineSuperTables.PREDICTIONS
+                ].super_table,
+                start=datetime.min,
+                end=datetime.now().astimezone(),
+            )
+
+        assert "effective_sample_count" in predictions_df.columns
+        assert "estimated_prediction_count" in predictions_df.columns
+        assert predictions_df.shape[0] == 20
+
+        predictions_df_with_sample = predictions_df[
+            predictions_df["endpoint_id"] == ep_id_with_sample
+        ]
+        predictions_df_without_sample = predictions_df[
+            predictions_df["endpoint_id"] == ep_id_without_sample
+        ]
+
+        # Validate that the model endpoint without sampling includes all the data points
+        assert predictions_df_without_sample["effective_sample_count"].sum() == 1000
+        assert predictions_df_without_sample["estimated_prediction_count"].sum() == 1000
+        # As for the model endpoint with sampling, the effective sample count should be around 155
+        # corresponding to the 15.5% sampling. We will validate that it is not equal to 1000.
+        assert predictions_df_with_sample["effective_sample_count"].sum() != 1000
```

Comment:
Should we verify that the `estimated_prediction_count` falls within a reasonable range?

Reply:
Don't think it's necessary because it's a statistcal sampling so I don't want to risk in test failures in these cases. Theoretically, the test could also fail with an effective sample size of 1000, but the probability of that happening is almost impossible.